_posts/2022-06-22-domain-management-for-multitenant-app-in-aws.markdown

```diff
@@ -0,0 +1,64 @@
+
+You might think it has more business value for you to focus on maintaining and developing the application, but managing the domain is critical in keeping it live and accessible. 
+
+Before we dive into the solution, what is a multitenant application? Multitenant application has a single instance of an application, and its infrastructure and business logic serves multiple clients. The book [Why Multi-Tenancy is Key to Successful and Sustainable Software-as-a-Service](https://books.apple.com/us/book/why-multi-tenancy-is-key-to-successful-and/id419723802) written by Larry Aiken explains the why it is essential to have multitenancy for some applications as compared to a single tenant.
+
+What makes multitenancy challenging, and so as domain management for multitenant? Multitenancy is a challenging problem because you must consider your application's security, optimization, monitoring, and high availability. Each aspect of implementation could be a real challenge when it grows.
+
+In this post, I would love to share my thoughts on how you can design your application to increase the availability, visibility and reduce some repetitive tasks for each client within the same application instance for domain management.
+
+### Solution
+
+It depends whether you are hosting your application on cloud or on premise. Cloud hosting has greater flexibility and scalability than on-premise hosting. Hence in this article, I will explain how you can implement your domain management system in AWS. 
+
+The diagram below is the solution to how you can typically start building it, but it will soon become challenging when the number of clients grows.
+
+![Domain Management Architecture ](/img/blog/domain-management/architectureSingleAccount.png)
+
+Generally, it is a good practice to segregate the production environment from the development environment, which should also apply to your domain management system. This approach has no segregation between environments and hosts all domain names within a single location. It increases the risk of impacting all tenants when there are service disruptions. Hence it has a greater chance of affecting the availability of the application. 
+
+The architecture diagram below shows how to manage domains for all tenants in multiple environments to minimize the likelihood of impacting your production environment.  
+
+
+![Domain Management Architecture ](/img/blog/domain-management/architectureDiagram.png)
+
+The critical difference between these two diagrams is the latter has a clear separation between environments. The infrastructure for the non-production environment should be identical to the production environment to enable errors to be caught in the non-production environment first without impacting the production domains. DNS is a public-facing element you would want to minimize downtime as it will directly affect your revenue or reputation.
```

Comment:
```suggestion
There is now a clear separation between environments. The non-production environment should be identical to the production environment to enable errors to be caught in the non-production environment first without impacting the production domains. DNS is a public-facing element: you should aim to minimise downtime as it may directly affect your revenue or reputation.
```

Proofreading suggestions.